tests/test_trivia.py

```diff
@@ -10,23 +11,63 @@ def test_QuestionList():
     for letter in "ABCDEFGHIJKLM":
         questions.append(QuestionAnswer(letter, letter))
     qlist = QuestionList(questions)
-    assert qlist.count == len(questions)
-    assert len(qlist) == len(questions)
-    for _ in range(25):
-        assert qlist.get_random_question() in questions
+    assert qlist._questions == questions
 
+def test_QuestionList_add_question():
+    qlist = QuestionList([])
     test_question = QuestionAnswer("test question", "test answer")
     qlist.add_question(test_question)
-    assert qlist.count == len(questions) + 1
-
-    questions2 = []
-    for letter in "NOPQRSTUVWXYZ":
-        questions2.append(QuestionAnswer(letter, letter))
-    qlist.add_questions(questions2)
-
-    all_questions = questions + questions2 + [test_question]
-    assert qlist.count == len(all_questions)
-    for q in all_questions:
-        assert q in qlist._questions
-    for q in qlist._questions:
-        assert q in all_questions
+    assert qlist._questions == [test_question]
+
+def test_QuestionList_add_questions():
+    qlist = QuestionList([])
+    questions = []
+    for letter in "ABCDEFGHIJKLM":
+        questions.append(QuestionAnswer(letter, letter))
+    qlist.add_questions(questions)
+    assert qlist._questions == questions
+
+def test_QuestionList_count_len():
+    questions = []
+    for letter in "ABCDEFGHIJKLM":
+        questions.append(QuestionAnswer(letter, letter))
+    qlist = QuestionList(questions)
+    assert len(qlist) == qlist.count == len(questions)
+
+def test_QuestionList_random_question():
+    test_question = QuestionAnswer("test question", "test answer")
+    qlist = QuestionList([test_question])
+    assert qlist.get_random_question() == test_question
+
+def test_load_trivia_questions():
+    questions = load_trivia_questions("tests/testquestions.txt")
+    expected = [
+        QuestionAnswer("Question A", "Answer A"),
+        QuestionAnswer("Question B", "Answer B"),
+        QuestionAnswer("Question C", "Answer C"),
+        QuestionAnswer("Question D", "Answer D"),
+    ]
+    assert questions == expected
+
+def test_Trivia():
+    trivia = Trivia()
+    assert isinstance(trivia.qlist, QuestionList)
+    assert trivia.inProgress is False
+
+def test_Trivia_start():
+    trivia = Trivia()
+    assert trivia.start() == 0
```

Comment:
If we were implementing this trivia command for real, it would be good to also test here whether or not trivia.inProgress is changed to true after the !start command

Reply:
Overall, your testing is very thorough and you put a lot of work into this!

Reply:
> If we were implementing this trivia command for real, it would be good to also test here whether or not trivia.inProgress is changed to true after the !start command

Indeed, but I can only test what's been implemented ;)